awx/api/serializers.py

```diff
@@ -2321,6 +2323,11 @@ class Meta:
         )
         read_only_fields = ('*', 'custom_virtualenv')
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if 'source' in self.fields:
+            self.fields['source'].choices = return_inventory_source_options() or {}
```

Comment:
Is there a case where `return_inventory_source_options()` would return a falsy object like `[]`, or `None`, or `''`? If not, then the fallback is not needed:
```suggestion
            self.fields['source'].choices = return_inventory_source_options()
```

Reply:
@chrismeyersfsu and I did chat about that, and I don't think there is a case for that. Given that, I will go ahead and commit this. Thank you!